.github/workflows/ci.yml

```diff
@@ -0,0 +1,37 @@
+name: Build Documentation using MkDocs
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the main branch
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
```

Comment:
Wouldn't the presence of `mkdocs gh-deploy --force` means the PR gets pushed before it has merged? Not sure, just asking.

Reply:
Looks like this is confined to only merges/PRs to the main branch. That said we should break this "build" job into a "build" and "deploy" phase and enable the build job for all branches/PRs.

Reply:
Agree on splitting it out

Reply:
Might also want to check these links re: getting `mkdocs gh-deploy` working from a github action:

https://github.com/mkdocs/mkdocs/discussions/2369
https://github.com/mkdocs/mkdocs/issues/2370
https://github.com/actions/checkout#fetch-all-history-for-all-tags-and-branches

Reply:
I am using the same job but step can be controlled based on branch ref name: https://docs.github.com/en/actions/reference/environment-variables so 
- Runs on any branch for push events.
- Runs on pull_request of `main` branch only
- if pull_request reports merge conflicts/ fails then it will not run deployment
- use the same jobs while pushing any branch but only triggers the deploy using `mkdocs gh-deploy --force` only when working on `main` branch.

Reply:
Another version I have is:
```
name: Build and Deploy Documentation using MkDocs

# Controls when the action will run. Triggers the workflow on push events on any branch 
# but pull request events only for the main branch
on:
  push: #  for any branch
    # branches:
    #   - '**'
  pull_request:
    branches: [ main ]
  
  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

jobs:
  setup: 
    name: Setup Ubuntu 
    runs-on: ubuntu-latest
    strategy:
      matrix:
        python-version: [ 3.8 ]

    steps:
      - name: Checkout Main
        uses: actions/checkout@v2

      - name: Set up Python ${{ matrix.python-version }}
        uses: actions/setup-python@v2
        with:
          python-version: ${{ matrix.python-version }}
          architecture: x64
  build:
    name: Build Documentation
    needs: setup
    runs-on: ubuntu-latest

    steps:    
      - name: Install dependencies
        run: |
          python -m pip install --upgrade pip
          pip install -r requirements.txt
          git config user.name 'github-actions[bot]' && git config user.email 'github-actions[bot]@users.noreply.github.com'
  deploy:
    name: Deploy Documentation
    needs: setup
    runs-on: ubuntu-latest

    steps:
      - name: Deploy
        # this part will only run if the PR has been merged
        # if: github.event.pull_request.merged == true
        # this part will only run if the branch or tag ref 
        # that triggered the workflow is main i.e. pushed or PR merged in main
        if: github.ref == 'refs/heads/main'
        run: |
          mkdocs gh-deploy --force

```

Reply:
Looks good, however, the pip  install and git config should happen int he setup job and the build step should just be running `mkdocs build`